overlord/assertstate/bulk.go

```diff
@@ -141,6 +142,26 @@ func bulkRefreshSnapDeclarations(s *state.State, snapStates map[string]*snapstat
 	return nil
 }
 
+func bulkRefreshRegistries(s *state.State, registries []*registry.Registry, userID int, deviceCtx snapstate.DeviceContext, opts *RefreshAssertionsOptions) error {
```

Comment:
This and the function below have a lot in common, might be something to consider making parametric, but that might be more annoying than it is worth.

Reply:
Maybe that's not the function you mean but `bulkRefreshValidationSetAsserts` has quite a bit of validation set specific logic. I think it might be hard to refactor them in a way worthwhile way